python/lib/dependabot/python/file_updater/pip_compile_file_updater.rb

```diff
@@ -223,6 +223,7 @@ def install_required_python
           return if run_command("pyenv versions").include?("#{python_version}\n")
 
           run_command("pyenv install -s #{python_version}")
+          run_command("pyenv exec pip install --upgrade pip")
```

Comment:
Just so I understand how this works... you're first manually saying grab the latest version of `pip` here, and then in the subsequent command is where it will revert back to a specific `pip` version?

The original reason for this PR was a broken `pip` version, so with this current code we could still end up installing a broken pip in the first step, before trying to revert it... but you decided not to pin here because then suddenly we're hardcoding versions within the code rather than the lockfile? 
And if a new version of `pip` ships that's broken, we can temp pin here and then revert once the latest version is unbroken?

I'm totally fine with the route you went, just want to confirm my understanding is correct.

Reply:
Yes, that is all correct.  With the additional detail that when pip ships a broken version (they do this at least a couple times a year in my experience), it probably won't be broken for a basic install like the second install here, so we should always be able to revert to the pinned version in the lock file.